crates/env/src/engine/off_chain/test_api.rs

```diff
@@ -355,3 +393,12 @@ pub fn assert_contract_termination<T, F>(
     assert_eq!(value_transferred, expected_value_transferred_to_beneficiary);
     assert_eq!(beneficiary, expected_beneficiary);
 }
+
+#[macro_export]
+/// Prepend contract message call with value transfer. Used for tests in off-chain environment.
+macro_rules! pay_with_call {
+            ($contract:ident . $message:ident ( $($params:ty)? ) , $amount:expr) => {{
+                ::ink_env::test::transfer_in::<Environment>($amount);
```

Comment:
```suggestion
                ::ink_env::test::transfer_in::<$crate::Environment>($amount);
```

Reply:
nope, this won't compile

we can leave it just `Environment` here because this type [is always declared](https://github.com/paritytech/ink/blob/ag-call-payable-offchain/crates/lang/codegen/src/generator/env.rs#L35) in the generated contract code.